### PR TITLE
WIP: Schedulers

### PR DESCRIPTION
**Pull request by dialelo**
- [ ] Add extra arities to functions that return schedulers

@niwinz are you ok with adding an extra arity to fns that can accept schedulers? I was thinking on adding it as a parameter in the end, like in Rx.

### CODE REVIEW

File: src/beicon/core.cljs [hunk 1]
```diff
@@ -508,6 +508,25 @@
...
+
+(defn scheduler?
+  [s]
+  (js/Rx.Scheduler.isScheduler s))
+
+(def immediate-scheduler js/Rx.Scheduler.immediate)
+(def current-thread-scheduler js/Rx.Scheduler.currentThread)
+
+(defn observe-on
+  [ob scheduler]
```
Comment (niwinz) [inline src/beicon/core.cljs]: The `ob` should be the last parameter I think!

**Comment by niwinz** [comment]
If it matters ok for me, but I'm not pretty convinced about that. Having the observable as last argument this will makes the scheduler as butlast argument and it seems a little bit ugly there..

**Comment by dialelo** [comment]
Ok, I'll add the extra arity with the scheduler before the observable as we discussed off-the-record.

File: assets/_rxjs/externs.js [hunk 2]
```diff
@@ -22,6 +21,32 @@ Rx.Subject = function() {};
...
+/**
+ * @const
+ */
+Rx.Scheduler.currentThread = function() {};
+
+/**
+ * @this {null}
+ * @return {boolean}
+ */
+Rx.Scheduler.prototype.isScheduler = function() {};
```
Comment (niwinz) [inline assets/_rxjs/externs.js]: This should be `Rx.Scheduler.isScheduler` I think.

File: assets/_rxjs/externs.js [hunk 3]
```diff
@@ -22,6 +21,32 @@ Rx.Subject = function() {};
 Rx.ConnectableObservable = function() {};
 
 /**
+ * @const
+ */
```
Comment (niwinz) [inline assets/_rxjs/externs.js]: "@constructor" annotation is missing.

**Comment by niwinz** [comment]
That is the state of this. Can I take it over?

**Comment by dialelo** [comment]
Yes please, go ahead. I forgot that I'd opened this PR :sweat:

**Comment by niwinz** [comment]
Already merged! \o/

**Pull request closed without merging**